### ISSUE TITLE
[REVIEW]: ppdiag: Diagnostic Tools for Temporal Point Processes

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3133]
**Submitting author:** @OwenWard (<a href="http://orcid.org/0000-0002-9678-3542">Owen G. Ward</a>)
**Repository:** <a href="https://github.com/OwenWard/ppdiag" target ="_blank">https://github.com/OwenWard/ppdiag</a>
**Version:** v0.1.1
**Editor:** @mikldk
**Reviewer:** @alexpghayes, @wjakethompson
**Archive:** <a href="https://doi.org/10.5281/zenodo.4774232" target="_blank">10.5281/zenodo.4774232</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/ca4ecdba18eb0b712e123ec47f6c3868/status.svg)](https://joss.theoj.org/papers/ca4ecdba18eb0b712e123ec47f6c3868)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/ca4ecdba18eb0b712e123ec47f6c3868"><img src="https://joss.theoj.org/papers/ca4ecdba18eb0b712e123ec47f6c3868/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/ca4ecdba18eb0b712e123ec47f6c3868/status.svg)](https://joss.theoj.org/papers/ca4ecdba18eb0b712e123ec47f6c3868)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/OwenWard/ppdiag" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@alexpghayes & @wjakethompson, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @mikldk know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/OwenWard/ppdiag">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@OwenWard) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/OwenWard/ppdiag">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@OwenWard) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @alexpghayes, @wjakethompson it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by mikldk** [comment]
@alexpghayes, @wjakethompson: Thanks for agreeing to review. Please carry out your review in this issue by updating the checklist above and giving feedback in this issue. The reviewer guidelines are available here: <https://joss.readthedocs.io/en/latest/reviewer_guidelines.html>. If possible create issues (and cross-reference) in the submission's repository to avoid too specific discussions in this review thread. 

If you have any questions or concerns please let me know.

**Comment by mikldk** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.39 s (316.0 files/s, 41872.4 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
HTML                            48           2296            638           7401
R                               54            272            924           2407
CSS                              3             99             48            428
Rmd                              3            194            324            336
JavaScript                       5             65             37            277
Markdown                         3             58              0            228
YAML                             5             32              2            155
XML                              1              0              0             63
SVG                              1              0              1             11
-------------------------------------------------------------------------------
SUM:                           123           3016           1974          11306
-------------------------------------------------------------------------------


Statistical information for the repository '8f3a7c16f23dd68666a63c8f' was
gathered on 2021/03/24.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Owen Ward                        3           415              0          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Owen Ward                   415          100.0          0.0               10.36
```

**Comment by mikldk** [comment]
@whedon check references from branch joss-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch joss-paper
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3133 with the following error: 

 Can't find any papers to compile :-(

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/08997660252741149 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by mikldk** [comment]
@whedon generate pdf from branch joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: :point_left:

**Comment by mikldk** [comment]
@alexpghayes, @wjakethompson: Thanks for agreeing to review. Please carry out your review in this issue by updating the checklist above and giving feedback in this issue. The reviewer guidelines are available here: <https://joss.readthedocs.io/en/latest/reviewer_guidelines.html>. If possible create issues (and cross-reference) in the submission's repository to avoid too specific discussions in this review thread. 

If you have any questions or concerns please let me know.

**Comment by alexpghayes** [comment]
I can't edit the checklist and my invitation to `joss-reviews` has expired. Can I get a new one @mikldk?

**Comment by mikldk** [comment]
@whedon re-invite @alexpghayes as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@alexpghayes please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by whedon** [comment]
:wave: @alexpghayes, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @wjakethompson, please update us on how your review is going (this is an automated reminder).

**Comment by wjakethompson** [comment]
I've completed my initial review. Just few minor comments/suggestions.

* __Community guidelines__: See OwenWard/ppdiag#54
* __References__: See OwenWard/ppdiag#55

Some other thoughts:

* There are authors on the paper that haven't contributed to the repo, but I went ahead and checked that box based on [this comment](https://github.com/openjournals/joss-reviews/issues/3085#issuecomment-791029381).
* I would generally prefer more test coverage than the 56% the package currently has. However, this isn't an acceptance blocker for me, because it looks like the core functionality is tested, and it looks like the non-tested files are generally visualization helpers and wrappers around other functions that are tested (e.g., `drawUniMMHPIntensity.R` has no coverage, but it appears that the main purpose of this function is to wrap `drawHPIntensity.R`, which does have some coverage).
* Finally, there is no explicit mention in the statement of need of the target audience, but it seems fairly clear to me to that the target audience would be researchers using temporal point processes, but not sure if it should be more specific or explicit.

**Comment by mikldk** [comment]
@alexpghayes, can you please give a brief status of your review? This is not to rush you, merely to give me an impression of the progress and time-frame.

**Comment by OwenWard** [comment]
Thank you very much for your feedback @wjakethompson, we will try to address these comments shortly.

**Comment by alexpghayes** [comment]
I have started my review but need to find a chunk of time to sit down and complete it. I anticipate this will happen within the next week.

**Comment by mikldk** [comment]
@alexpghayes, @wjakethompson, can you please give a brief status of your reviews? This is merely to give me an impression of the progress and time-frame.

**Comment by alexpghayes** [comment]
Also just finished my review. On the whole seems like a well written and useful package. I agree with @wjakethompson that a `CONTRIBUTING.md` file would be helpful.

Other minor feedback:

- It would be nice to provide `print()` methods for `hp` objects, etc
- `pp_diag()` is currently an odd mixture of returning information, and causing side effects (printing as at https://github.com/OwenWard/ppdiag/blob/main/R/pp_diag.R#L46) and plotting. This pattern can be hard for future work to extend upon. I recommend either: (1) creating a diagnostics object that users can interact with programmatically, or (2) returning something like `NULL` and only causing side effects in `pp_diag()`.
- I'm pretty sure that if you use `@keywords internal` rather than `@keywords Internal` you won't need to use `@noRd` when documenting internal functions. Current use is I believe redundant (see https://github.com/OwenWard/ppdiag/blob/main/R/negloglik.R#L10-L11 for example)
- Not that it really matters, but running `styler` once on the package to clean up whitespace, and removing `.DS_Store` files (and adding `.DS_Store` to `.gitignore`) would be minor stylistic improvements.

**Comment by OwenWard** [comment]
@alexpghayes thanks for the helpful feedback also. We'll address these useful comments shortly!

**Comment by wjakethompson** [comment]
> @alexpghayes, @wjakethompson, can you please give a brief status of your reviews? This is merely to give me an impression of the progress and time-frame.

I believe my review is complete. There is still one outstanding PR (OwenWard/ppdiag#55) and one open issue (OwenWard/ppdiag#54), which @OwenWard has indicated the team will be addressing soon. Once those are closed I will check those boxes in my checklist. Or I can go ahead and check them now if that's the preferred process.

**Comment by mikldk** [comment]
@alexpghayes Thanks.

@wjakethompson Thanks. Please wait checking the boxes until the PRs are merged.

@OwenWard Please let @wjakethompson know when you have addressed the issues.

**Comment by OwenWard** [comment]
@wjakethompson @alexpghayes thanks for providing issues and pull requests along with the other feedback. We've closed the opened issues along with using the feedback to clean up some functions and add to the statement of need in the paper.
We really appreciate all the thoughtful feedback and suggestions!

@mikldk please let me know if there is anything else

**Comment by mikldk** [comment]
I believe the review is finished once @wjakethompson checks the last few boxes. Once this is done, I will ask the reviewers if they have finished their review.

**Comment by wjakethompson** [comment]
@mikldk things look good to me! I have finished my checklist.

**Comment by mikldk** [comment]
@alexpghayes also misses a few boxes -- can you please check all is okay and confirm here when you have finished your review?

**Comment by alexpghayes** [comment]
Will do. I have limited internet until Friday and anticipate completing my
review then.

**Comment by alexpghayes** [comment]
Just took a look, everything seems good to me!

**Comment by mikldk** [comment]
@OwenWard:

* Please have a final read though of the paper, checking language etc.
* Have a final check of the proofs with `@whedon generate pdf`
* In general, be sure that all versions are correct and that the repo is in the state that should be published.
* Please make a tagged release and archive (e.g. with [Zenodo](https://zenodo.org/)) as described [here](<https://joss.readthedocs.io/en/latest/submitting.html#the-review-process>), and report the version number and archive DOI in this thread. Please verify that the archive deposit has the correct metadata (title and author list), or edit these if that is not the case.

**Comment by OwenWard** [comment]
@whedon generate pdf from branch joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: :point_left:

**Comment by mikldk** [comment]
@whedon set v0.1.1 as version

**Comment by whedon** [comment]
OK. v0.1.1 is the version.

**Comment by mikldk** [comment]
@whedon set 10.5281/zenodo.4774232 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4774232" target="_blank">10.5281/zenodo.4774232</a> is the archive.

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: :point_left:

**Comment by mikldk** [comment]
@whedon check references

**Comment by mikldk** [comment]
@whedon check references from branch joss-paper

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/08997660252741149 is OK

MISSING DOIs

- 10.1002/9781119821588.ch7 may be a valid DOI for title: Diagnostics and Visualization of Point Process Models for Event Times on a Social Network

INVALID DOIs

- None
```

**Comment by mikldk** [comment]
@OwenWard Can you add `doi` for `PtProcess2010` reference - maybe it is `10.18637/jss.v035.i08`?

**Comment by OwenWard** [comment]
@mikldk done

**Comment by mikldk** [comment]
@whedon check references from branch joss-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch joss-paper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/08997660252741149 is OK
- 10.18637/jss.v035.i08 is OK

MISSING DOIs

- 10.1002/9781119821588.ch7 may be a valid DOI for title: Diagnostics and Visualization of Point Process Models for Event Times on a Social Network

INVALID DOIs

- None
```

**Comment by mikldk** [comment]
@whedon generate pdf from branch joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: :point_left:

**Comment by mikldk** [comment]
@OwenWard: It also seems that the arXiv paper is now published at [10.1002/9781119821588.ch7](https://doi.org/10.1002/9781119821588.ch7). Can you either confirm or refute that?

**Comment by OwenWard** [comment]
@mikldk correct, I'll update to that reference now!

**Comment by OwenWard** [comment]
@mikldk updated

**Comment by mikldk** [comment]
@whedon generate pdf from branch joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: :point_left:

**Comment by mikldk** [comment]
@whedon check references from branch joss-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch joss-paper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/08997660252741149 is OK
- 10.18637/jss.v035.i08 is OK

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.1002/9781119821588.ch7 is INVALID because of 'https://doi.org/' prefix
```

**Comment by mikldk** [comment]
> @mikldk updated

Not properly. Please check paper; there's an extra `https...` prefix in the References.

**Comment by OwenWard** [comment]
@whedon check references from branch joss-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch joss-paper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/08997660252741149 is OK
- 10.18637/jss.v035.i08 is OK
- 10.1002/9781119821588.ch7 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by OwenWard** [comment]
@mikldk corrected now, sorry about that

**Comment by mikldk** [comment]
@whedon check references from branch joss-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch joss-paper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/08997660252741149 is OK
- 10.18637/jss.v035.i08 is OK
- 10.1002/9781119821588.ch7 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by mikldk** [comment]
@whedon generate pdf from branch joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03133/joss.03133/10.21105.joss.03133.pdf) :page_facing_up: :point_left:

**Comment by mikldk** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3133 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by mikldk** [comment]
@whedon accept from branch joss-paper

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/08997660252741149 is OK
- 10.18637/jss.v035.i08 is OK
- 10.1002/9781119821588.ch7 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2339

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2339, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss-paper 
```

**Comment by arfon** [comment]
@OwenWard – please merge this PR which makes some small changes to your paper (the metadata ones are essential for compatibility with our final publishing workflow steps): https://github.com/OwenWard/ppdiag/pull/59

**Comment by OwenWard** [comment]
@arfon done

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1162/08997660252741149 is OK
- 10.18637/jss.v035.i08 is OK
- 10.1002/9781119821588.ch7 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2342

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2342, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss-paper 
```

**Comment by arfon** [comment]
@whedon accept deposit=true from branch joss-paper

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1397897852458881024) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2343
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03133](https://doi.org/10.21105/joss.03133)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@alexpghayes, @wjakethompson – many thanks for your reviews here and to @mikldk for editing this submission! JOSS relies upon the volunteer efforts of folks like yourselves, and we simply wouldn't be able to do this without you ✨ 

@OwenWard – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03133/status.svg)](https://doi.org/10.21105/joss.03133)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03133">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03133/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03133/status.svg
   :target: https://doi.org/10.21105/joss.03133
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03133/status.svg)](https://doi.org/10.21105/joss.03133)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by OwenWard** [comment]
Thank you @wjakethompson @alexpghayes @mikldk so much for reviewing and editing. This has been an excellent process and I look forward to being involved in it again on both the submission and reviewing side.

Thanks @arfon!
